entities/StandardBoard.cs

```diff
@@ -0,0 +1,176 @@
+
+        // Dictionary to save pieces by cell.
+        public Dictionary<Cell, Piece> PositionToPiece;
+
+        public StandardBoard()
+        {
+            PositionToPiece = new Dictionary<Cell, Piece>();
+        }
+
+        public StandardBoard(Dictionary<Cell, Piece> positionToPiece)
+        {
+            // Deep copy of dictionary.
+            PositionToPiece = positionToPiece.ToDictionary(entry => entry.Key,
+                                                           entry => entry.Value);
+        }
+
+        /// <summary>
+        /// Deep copy of the standard board.
+        /// </summary>
+        /// <returns>
+        /// The board with pieces equal to the current/this board. 
+        /// </returns>
+        public StandardBoard DeepCopy()
+        {
+            return new StandardBoard(PositionToPiece);
+        }
+
+        /// <summary>
+        /// Checks if the cell is valid.
+        /// </summary>
+        /// <exception 
+        ///cref="ArgumentOutOfRangeException">Invalid coordinates.
+        ///</exception>
+        private void CheckCell(Cell cell)
+        {
+            if (!OnBoard(cell))
+            {
+                throw new ArgumentOutOfRangeException($"Invalid cell coordinates {cell.X} and {cell.Y}");
+            }
+        }
+
+        /// <summary>
+        /// Checks if the cell is on board.
+        /// </summary>
+        /// <param name="cell">
+        /// The cell to check.
+        /// </param>
+        /// <returns>
+        /// true if the cell is on board, otherwise false. 
+        /// </returns>
+        public bool OnBoard(Cell cell)
+        {
+            return 0 <= cell.X &&
+                   cell.X < Width &&
+                   0 <= cell.Y &&
+                   cell.Y < Height;
+        }
+
+        /// <summary>
+        /// Checks if the cell contains any pieces.
+        /// </summary>
+        /// <param name="cell">
+        /// The cell.
+        /// </param>
+        /// <returns>
+        /// true if the cell contains a piece, otherwise false. 
+        /// </returns>
+        public bool IsEmpty(Cell cell)
+        {
+            CheckCell(cell);
+            return !PositionToPiece.ContainsKey(cell);
+        }
+
+        /// <summary>
+        /// Find piece by cell.
+        /// </summary>
+        /// <param name="cell">
+        /// The cell.
+        /// </param>
+        /// <returns>
+        /// piece if exists, otherwise null. 
+        /// </returns>
+        public Piece? GetPiece(Cell cell)
+        {
+            CheckCell(cell);
+            if (PositionToPiece.ContainsKey(cell))
+                return PositionToPiece[cell];
+            return null;
+        }
+
+        /// <summary>
+        /// Set piece at cell.
+        /// </summary>
+        /// <param name="cell">
+        /// The cell.
+        /// </param>
+        /// <param name="piece">
+        /// The piece.
+        /// </param>
+        public void SetPiece(Cell cell, 
```

Comment:
Why are parameters placed on different lines? I believe there is enough space for both parameters.